stub/src/main/java/io/grpc/stub/StreamObservers.java

```diff
@@ -44,13 +44,18 @@
     Preconditions.checkNotNull(target, "target");
 
     final class FlowControllingOnReadyHandler implements Runnable {
+      private boolean completed;
+
       @Override
       public void run() {
-        while (target.isReady() && source.hasNext()) {
-          target.onNext(source.next());
-        }
-        if (!source.hasNext()) {
-          target.onCompleted();
+        if (!completed) {
```

Comment:
nit: Instead, do the following to avoid increasing the indentation.

```java
if (completed) {
  return;
}
```

Reply:
👍 updated